Demo/HYPForms/HYPSampleCollectionViewController.m

```diff
@@ -29,13 +30,21 @@ @implementation HYPSampleCollectionViewController
 - (instancetype)initWithDictionary:(NSDictionary *)dictionary
 {
     HYPFormsLayout *layout = [[HYPFormsLayout alloc] init];
-
     self = [super initWithCollectionViewLayout:layout];
     if (!self) return nil;
 
-    layout.dataSource = self;
+    self.layout = layout;
     self.initialValues = dictionary;
 
+    [self.collectionView registerClass:[HYPImageFormFieldCell class]
+            forCellWithReuseIdentifier:HYPImageFormFieldCellIdentifier];
+
+    self.collectionView.dataSource = self.dataSource;
+
+    if ([NSObject isUnitTesting]) {
+        [self.collectionView numberOfSections];
```

Comment:
It's weird, but initializing the controller in a unit test doesn't trigger any of the dataSource methods:

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView;
- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section;
So the next time these are called is when a change has been made to it's data, so the dataSource no longer contains the correct state.

That's why this weird thing is needed.